torchao/utils.py

```diff
@@ -110,61 +109,6 @@ def flat_size(tensor):
             model_size += get_model_size_in_bytes(child, ignore_embeddings)
     return model_size
 
-class UnwrapTensorSubclass(torch.nn.Module):
-    def forward(self, *tensors):
-        todo = list(tensors)
-        for tp, meta, inner_tensors in reversed(self.rebuild_stack):
-            nb_tensor = len(inner_tensors)
-            inner_tensors = {a: b for a, b in zip(inner_tensors, todo[-nb_tensor:])}
-            todo = todo[nb_tensor:]
-            rebuilt = tp.__tensor_unflatten__(inner_tensors, meta, None, None)
-            todo.append(rebuilt)
-
-        assert len(todo) == 1
-        return todo[0]
-
-    def right_inverse(self, tensor):
-        assert type(tensor) is not torch.Tensor
-        rebuild_stack = []
-        plain_tensors = []
-        todo = [tensor]
-        while todo:
-            obj = todo.pop()
-            inner_tensors, metadata = obj.__tensor_flatten__()
-            rebuild_stack.append((type(obj), metadata, inner_tensors))
-            for attr_name in inner_tensors:
-                val = getattr(obj, attr_name)
-                if type(val) is torch.Tensor:
-                    plain_tensors.append(val)
-                else:
-                    assert isinstance(val, torch.Tensor)
-                    todo.append(val)
-
-        self.rebuild_stack = rebuild_stack
-
-        return plain_tensors
-
-def unwrap_tensor_subclass(model, filter_fn=None):
-    """Unwraps (nested) tensor subclass in the model to plain tensors
-    This is a workaround to make a model with tensor subclass to work with `torch.export.export`
-    and `torch.aot_compile`, we hope this can be integrated into compile stack soon
-    tracking issue: https://github.com/pytorch/ao/issues/345
```

Comment:
oh @jerryzh168 I think you still need this logic.

To clarify, there are two separate PRs/issues:

(1) nested subclass support here: https://github.com/pytorch/pytorch/pull/127431. This PR allows for "nested wrapper subclasses at the graph boundaries", so e.g. we now support DTensor(Float8Tensor) as a graph input to `torch.compile`).

(2) what I think is relevant to this PR, is that we eventually want to take the parametrization logic you have here and upstream it directly into export so it can happen automatically and you don't need to do it manually before exporting / running AOTI. This is not currently fixed, but I recently filed a tracker issue for it (with a float8 repro) here: https://github.com/pytorch/pytorch/issues/129682. @tugsbayasgalan is going to be picking it up soon.

Reply:
ah I see, OK I can wait until https://github.com/pytorch/pytorch/issues/129682 is resolved as well